scripts/make_triplets.py

```diff
@@ -0,0 +1,215 @@
+import argparse
+import collections
+import gzip
+import json
+import logging
+import os
+
+import tqdm
+import numpy as np
+
+import s2and
+from s2and.data import ANDData
+from s2and.consts import CONFIG
+from s2and.text import counter_jaccard, cosine_sim, STOPWORDS
+
+logger = logging.getLogger(__name__)
+
+DATASETS = [
+    "aminer",
+    "arnetminer",
+    "inspire",
+    "kisti",
+    "pubmed",
+    "qian",
+    "zbmath",
+]
+
+
+class NgramJaccardRanker:
+    def __init__(self, papers):
+        self.paper_ngrams = {k: self._unigrams(v["title"]) + self._unigrams(v["abstract"]) for k, v in papers.items()}
+
+    def _unigrams(self, text):
+        if text is None or len(text) == 0:
+            return collections.Counter()
+        text_split = [word for word in text.split() if word not in STOPWORDS and len(word) > 1]
+        unigrams = collections.Counter(text_split)
+        return unigrams
+
+    def __call__(self, sig1, sig2):
+        paper1 = self.paper_ngrams[str(sig1.paper_id)]
+        paper2 = self.paper_ngrams[str(sig2.paper_id)]
+
+        return counter_jaccard(paper1, paper2)
+
+
+class SpecterRanker:
+    def __init__(self, specters):
+        self.paper2idx = {paper_id: idx for idx, paper_id in enumerate(specters[1])}
+        self.specter = specters[0]
+
+    def __call__(self, sig1, sig2):
+        emb1 = self.specter[self.paper2idx[str(sig1.paper_id)]]
+        emb2 = self.specter[self.paper2idx[str(sig2.paper_id)]]
+
+        return cosine_sim(emb1, emb2)
+
+
+def generate_block_triplets(
+    dataset, block_sigs, rng, num_queries=None, num_positives=None, num_negatives=None, negative_ranker_fn=None
```

Comment:
Default `=None` will cause errors. Maybe just have no defaults or put in some sensible defaults.